internal/podtemplatespec/podtemplatespec.go

```diff
@@ -0,0 +1,119 @@
+package podtemplatespec
+
+import (
+	"fmt"
+
+	"github.com/kanopy-platform/hedgetrimmer/internal/limitrange"
+	"github.com/kanopy-platform/hedgetrimmer/internal/quantity"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	defaultLimitRequestMemoryRatio float64 = 1.1
+)
+
+func New(p corev1.PodTemplateSpec) *PodTemplateSpec {
+	pts := &PodTemplateSpec{
+		pts: *p.DeepCopy(),
+	}
+	return pts
+}
+
+func (p *PodTemplateSpec) ApplyResourceRequirements(lri corev1.LimitRangeItem) (corev1.PodTemplateSpec, error) {
```

Comment:
Why split the PodTemplateSpec into the constructor?   This would require each specific handler (Deployment, Statefulset) to create a new instance of this per mutation.

The `controller` will define a `PodTemplateSpecResourceMutator` interface that includes `ApplyResourceRequirements`.  

This library implements that interface.  The controller only needs to create one instance of the struct and pass that via DI (dependency injection) into each mutation handler.

Reply:
Ah ok. Moved the PodTemplateSpec back to the `Mutate` function call input argument and left the struct empty.